tool/build/apelink.c

```diff
@@ -2206,6 +2240,20 @@ int main(int argc, char *argv[]) {
     }
 
     // extract the ape loader for non-input architectures
+    // if the user requested a host kernel check, get the host kernel
+    if (loader_kernel[0] != '\0') {
```

Comment:
[optional; nit] My personal preference is to avoid redundancy and embrace the type messiness.